src/Evolution/Systems/ScalarAdvection/VelocityField.cpp

```diff
@@ -0,0 +1,48 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/ScalarAdvection/VelocityField.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace ScalarAdvection::Tags {
+
+template <size_t Dim>
+void VelocityFieldCompute<Dim>::function(
+    const gsl::not_null<tnsr::I<DataVector, Dim>*> velocity_field,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& inertial_coords) noexcept {
```

Comment:
@nilsdeppe After fixing both AddComputeTags and coordinates tags problem, compilation goes well while I got the following runtime error:
```
Line: 278 of /home/yoonsoo/work/spectre/src/DataStructures/VectorImpl.hpp
'rhs.size() == size()' violated!
Function: VectorImpl<T, VectorType> &VectorImpl<double, DataVector>::operator=
    (const VectorImpl<T, VectorType> &) [T = double, VectorType = DataVector]
Must copy into same size, not 0 into 5
############ ASSERT FAILED ############
```

So I tracked down a bit and found that, when `VelocityFieldCompute<Dim>::function` is called `velocity_field` has zero length while `inertial_coords` has correct length (=InitialGridPoints of input file). Do you have any idea why `velocity_field` has zero length?

Reply:
it's supposed to have zero length and then it gets resized inside the function. This is done with every compute tag. Without seeing any code I can't help you debug what your code is doing differently. Maybe you're missing a `destructive_resize_components(velocity_field, get<0>(inertial_coords).size())` call?

Reply:
Oh... I had no idea I was supposed to do resizing inside a compute tag 🙃 (because always have fed pre-sized vector when writing tests). Your guess is right, now it's working well.

Reply:
ah, sorry I didn't make that clear before! If you search for `destructive_resize` and `destructive_resize_components` you'll see that all the `not_null` functions call this at the start :) I'm glad to hear it's working now!